projects/Mallard/src/authentication/services/__tests__/apple.spec.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    AppleAuthenticationCredential,
+    AppleAuthenticationFullName,
+    AppleAuthenticationUserDetectionStatus,
+} from 'expo-apple-authentication'
+
+import { mapCredentials } from '../apple'
+
+describe('apple authorizer service', () => {
+    describe('mapCredentials', () => {
+        it('shoudl map apple credentials to Guardian credentials', () => {
+            const exampleAppleCredentials: AppleAuthenticationCredential = {
+                user: '111222333444',
+                state: 'some state',
+                fullName: {
+                    givenName: 'Jonjoe',
+                    familyName: 'Whitfield',
+                } as AppleAuthenticationFullName,
+                email: 'me@jonjoe.io',
+                realUserStatus: {} as AppleAuthenticationUserDetectionStatus,
+                identityToken: '',
+                authorizationCode: 'auth code',
+            }
+
+            const actualResult = mapCredentials(exampleAppleCredentials)
+            const expectedResult = {
+                authorizationCode: 'auth code',
+                givenName: 'Jonjoe',
+                familyName: 'Whitfield',
+                idToken: '111222333444',
+            }
+
+            expect(actualResult).toBe(expectedResult)
+        })
+    })
```

Comment:
could you add a test for a non-ideal scenario - e.g. when idToken is missing or something?